### ISSUE TITLE
UseContentfulLiveUpdates hook changes my object casing when live preview is enabled

### ISSUE THREAD

**Issue by betown** [contentful/live-preview#423]
**Describe the bug**
When using the hook useContentfulLiveUpdates to enable live previewing on the editor with the side-to-side view the name casing of my linked components within the RTE is changed from camelCase to snake_case causing the component to be unable to render the corresponding component content.

All the IDs on my data model are set as camelCase and they work most of the time even when passing through the hook function on the new tab preview, however, only during the side-to-side preview, the first level of keys (from outside to inside) is changed after going through the hook function, this seems to be affecting only the components referenced through the RTE as the rest of the references (Regular array field) seem to work just fine.

**To Reproduce**
1. Use camelCase to identify some components of the content model
2. Add those components through any RTE component as linked references
3. Use GraphQL to fetch the product data
4. Run the data through the hook **useContentfulLiveUpdates**
5. Log the data before and after the hook
6. If you are using the side-to-side preview notice how the data is changed
7. if you are using the new tab preview notice how it works as intended

**Expected behavior**
The data should remain consistent no matter where I am previewing the page I am working on

**Screenshots / example**
![image](https://github.com/contentful/live-preview/assets/30836562/18367315-19a0-40e9-9cc9-6574c87f9f81)

![image](https://github.com/contentful/live-preview/assets/30836562/48975a7c-0b10-4599-85c0-909779426992)

**Context (please complete the following information):**
 - Live preview SDK version number: 2.12.0
 - GraphQL API 

**Additional context**
* I don't have inspector mode set up on my code

**Comment by YvesRijckaert** [comment]
Hi @betown sorry for the late reply.
I tried reproducing this but was unable to. Do you mind sharing some additional context (your GraphQL Query, how you implement the useLiveUpdates hook etc). A minimal reproducible example on codesandbox would be ideal.

Here I tried reproducing it with a Post content type that has a rich text field and embeds 3 entries (block embed, inline embed and hyperlink embed) all of them had the correct field id's (camelCase) before and after making updates.
<img width="1396" alt="Screenshot 2024-01-05 at 09 42 29" src="https://github.com/contentful/live-preview/assets/22968325/706bca2b-0ca8-475e-9655-54bf5d09edaa">

**Comment by betown** [comment]
Hi @YvesRijckaert happy new year and sorry for the delay 

I finally managed to create a code sandbox with a copy of the website so you can check on the way I am implementing the live preview, hopefully, it will help to point out if/where I might be doing something different on the implementation.

I have created a new link (`/project/test-project`) that you'll be able to review within the code sandbox to confirm that they're working as intended on the live preview (you'll find the comment I set for the screenshot reference above is still there as well)

All my Graphql requests can be found on the following files `/lib/api.js` and `/lib/fragments.js` and the layout where i noticed the issue is on `/pages/project/[slug].js` you will notice at line `71` that there's a reference to the component `RTE` which is the one I'm trying to use to render all the components.

Feel free to let me know if you need any extra information.

Thanks in advance!

**Comment by YvesRijckaert** [comment]
Hi @betown 
Just wanted to say thanks for sending over that codesandbox link - I was able to spot and replicate the issue you mentioned.
Good news! We're gearing up for a new SDK release packed with bug fixes and some neat performance boosts. I tried out your project with our latest alpha release (`3.0.0-alpha.16`), and it looks like the issue is sorted.
We're planning to roll out this alpha officially in the next few days.
Cheers,
Yves
